Sources/Authenticator/Authenticator.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  Authenticator.swift
+//  Comet
+//
+//  Created by Tuan Tu Do on 26.05.2021.
+//  Copyright © 2021 Etnetera. All rights reserved.
+//
+
+import Combine
+import Foundation
+
+final class Authenticator {
+    private let tokenProvider: TokenProviding
+    private let queue = DispatchQueue(label: "Authenticator-\(UUID().uuidString)")
+    private var refreshTokenPublisher: AnyPublisher<String, AuthenticatorError>?
+
+    init(tokenProvider: TokenProviding) {
+        self.tokenProvider = tokenProvider
+    }
+
+    var accessToken: AnyPublisher<String, AuthenticatorError> {
+        queue.sync { [weak self] in
+            guard let unwrappedSelf = self else {
```

Comment:
You can simply call it `self`. Seems to be the latest convention.
```suggestion
            guard let self = self else {
```

Reply:
Yeah, I know. Will definitely change it. I just like to call it `unwrappedSelf` in the beginning of the implementation, because people tend to forget to capture the `self` weakly. And calling it `self` does not help at all. This is a way how I make sure that I won't create any retain cycles.